tests/e2e/scenarios/test_reborn_webui_v2_projects_api.py

```
@@ -0,0 +1,222 @@
+    return {
+        "id": body["user"]["user_id"],
+        "token": body["api_token"],
+    }
+
+
+def _project_ids(response: httpx.Response) -> set[str]:
+    assert response.status_code == 200, response.text
+    return {project["project_id"] for project in response.json()["projects"]}
+
+
+async def test_project_lifecycle_membership_and_restart_persistence_served(
```

Ruff (0.16.0)

[warning] Too many statements (85 > 50)

(PLR0915)